Sources/PrimerSDK/Classes/Core/Primer/Primer.swift

```diff
@@ -200,6 +210,117 @@ public class Primer {
             self?.root?.dismiss(animated: true, completion: nil)
         }
     }
+    // swiftlint:disable cyclomatic_complexity
+    public func performThreeDS(paymentMethod: PaymentMethodToken, completion: @escaping (Error?) -> Void) {
+        let state: AppStateProtocol = DependencyContainer.resolve()
+        
+        guard let clientToken = state.decodedClientToken else {
+            return completion(PrimerError.vaultFetchFailed)
+        }
+        
+        let service = ThreeDSecureService()
+        service.initializeSDK { (initResult) in
+            switch initResult {
+            case .success:
+                service.verifyWarnings { (verifyResult) in
+                    switch verifyResult {
+                    case .success:
+                        service.netceteraAuth(paymentMethod: paymentMethod, protocolVersion: .v1) { (authResult) in
+                            switch authResult {
+                            case .success(let transaction):
+                                let threeDSecureAuthData = try! transaction.buildThreeDSecureAuthData()
+                                print("3DS SDK Data: \(threeDSecureAuthData)")
+                                
+                                var req = ThreeDS.BeginAuthRequest.demoAuthRequest
+                                req.device = threeDSecureAuthData
+                                req.amount = 1000
```

Comment:
[Q]
What should be the logic here?

Reply:
As before, this is only because we're still testing and we have test cases. I expect all these to be fixed after integrating the decision logic. By the way, you have mentioned that we should be able to perform 3DS when vaulting, however the amount param on the request is required.

Reply:
In this case, I think we can put a 0 amount :)